packages/validator/src/services/attestation.ts

```diff
@@ -64,41 +65,86 @@ export class AttestationService {
     await Promise.race([sleep(this.clock.msToSlot(slot + 1 / 3), signal), this.emitter.waitForBlockSlot(slot)]);
     this.metrics?.attesterStepCallProduceAttestation.observe(this.clock.secFromSlot(slot + 1 / 3));
 
-    // Beacon node's endpoint produceAttestationData return data is not dependant on committeeIndex.
+    if (this.opts?.disableAttestationGrouping) {
+      // Attestation service grouping optimization must be disabled in a distributed validator cluster as
+      // middleware clients such as Charon (https://github.com/ObolNetwork/charon) expect the actual committee index
+      // to be sent to produceAttestationData endpoint. This is required because the middleware client itself
+      // calls produceAttestationData on the beacon node for each validator and there is a slight chance that
+      // the `beacon_block_root` (LMD GHOST vote) changes between calls which would cause a conflict between
+      // attestations submitted by Lodestar and other VCs in the cluster, resulting in aggregation failure.
+      // See https://github.com/ChainSafe/lodestar/issues/5103 for further details and references.
+      const dutiesByCommitteeIndex = groupAttDutiesByCommitteeIndex(duties);
+      await Promise.all(
+        Array.from(dutiesByCommitteeIndex.entries()).map(([index, duties]) =>
+          this.runAttestationTasksPerCommittee(duties, slot, index, signal).catch((e) => {
+            this.logger.error("Error running attestation tasks", {slot, index}, e);
```

Comment:
errors thrown by `runAttestationTasksPerCommittee` needs to be caught to avoid unhandled promise rejections and properly log all errors if multiple runs fails

this is also inline with the previous implementation

https://github.com/ChainSafe/lodestar/blob/106a05a281fa63b0d1b26571a573c7d46545626b/packages/validator/src/services/attestation.ts#L65-L67